lib/segment/src/types.rs

```diff
@@ -1370,6 +1371,69 @@ pub struct Range<T> {
     pub lte: Option<T>,
 }
 
+pub fn try_parse_datetime(s: &str) -> Option<DateTimePayloadType> {
+    // Attempt to parse the input string in RFC 3339 format
+    if let Ok(datetime) = chrono::DateTime::parse_from_rfc3339(s) {
+        return Some(datetime.into());
+    }
+
+    // Attempt to parse the input string in the specified formats
+    // YYYY-MM-DD'T'HH:MM:SS (without timezone or Z)
+    // YYYY-MM-DD HH:MM:SS
+    // YYY-MM-DD HH:MM
+    // YYYY-MM-DD
+    let datetime =
+        if let Ok(naive) = chrono::NaiveDateTime::parse_from_str(s, "%Y-%m-%d'T'%H:%M:%S") {
```

Comment:
The `T` should not be in quotes. It was only in quotes in the original list of formats to denote it as literal, and distinguish it from the other letters.

```suggestion
        if let Ok(naive) = chrono::NaiveDateTime::parse_from_str(s, "%Y-%m-%dT%H:%M:%S") {
```